autoscheduler/frontend/src/components/SchedulingPage/Schedule/MeetingCard/MeetingCard.tsx

```diff
@@ -3,32 +3,157 @@ import { Typography } from '@material-ui/core';
-  const handleResize = React.useCallback((newVal: boolean): void => {
-    setIsBig(newVal);
+  // Determines what size the card is, depending on the difference between the desired and available
+  // height.
+  const handleResize = React.useCallback((contentHeight: number, clientHeight: number): void => {
+    const heightDiff = clientHeight - contentHeight;
+    if (heightDiff > 30) {
+      setCardSize(MeetingCardSizes.large);
+    } else if (heightDiff > -15) { // Just enough for two rows of text
+      setCardSize(MeetingCardSizes.medium);
+    } else {
+      setCardSize(MeetingCardSizes.small);
+    }
   }, []);
 
+  const fullscreenLargeCard = (
+    <>
+      <Typography variant="body2" className={styles.meetingCardText}>
+        {`${section.subject} ${section.courseNum}`}
+        {`-${section.sectionNum}`}
+        &nbsp;
+        &bull;
+        &nbsp;
+        <Typography variant="subtitle2" component="span">
+          {`${MeetingType[meetingType]}`}
+        </Typography>
+      </Typography>
+      <Typography variant="subtitle2">
+        {meetingBuildingWithRoom(meeting)}
+      </Typography>
+      <Typography variant="body2" className={styles.meetingCardText}>
+        {meetingTimeText(meeting)}
+      </Typography>
+    </>
+  );
+
+  const fullscreenMediumCard = (
+    <>
+      <Typography variant="body2" className={styles.meetingCardText}>
+        {`${section.subject} ${section.courseNum}`}
+        {`-${section.sectionNum}`}
+        &nbsp;
+        &bull;
+        &nbsp;
+        <Typography variant="subtitle2" component="span">
+          {`${MeetingType[meetingType]}`}
+        </Typography>
+      </Typography>
+      <Typography variant="body2" className={styles.meetingCardText}>
+        <>
+          <Typography variant="subtitle2" component="span">
+            {meetingBuildingWithRoom(meeting)}
+          </Typography>
+          &nbsp;
+          &bull;
+          &nbsp;
+        </>
+        {meetingTimeText(meeting)}
+      </Typography>
+    </>
+  );
+
+  const fullscreenSmallCard = (
+    <>
+      <Typography variant="body2" className={styles.meetingCardText}>
+        {`${section.subject} ${section.courseNum}`}
+        {`-${section.sectionNum}`}
+        &nbsp;
+        &bull;
+        &nbsp;
+        <Typography variant="subtitle2" component="span">
+          {meetingBuildingWithRoom(meeting)}
+        </Typography>
+        &nbsp;
+        &bull;
+        &nbsp;
+        {meetingTimeText(meeting)}
+      </Typography>
+    </>
+  );
+
+  // Determine which size of fullscreen card to use
+  let fullscreenCard = null;
+  switch (cardSize) {
+    case MeetingCardSizes.large:
+      fullscreenCard = fullscreenLargeCard;
+      break;
+    case MeetingCardSizes.medium:
+      fullscreenCard = fullscreenMediumCard;
+      break;
+    case MeetingCardSizes.small:
+      fullscreenCard = fullscreenSmallCard;
+      break;
+    default:
+      break;
+  }
+
+  const normal = (
```

Comment:
Instead of having `normal` and `fullscreenCard` variables, just have a single `card` variable that determines what card type to use

Reply:
I will do it now, but just want to note that there is a chance that I have to re-do this change in save schedules as image, as at the moment I render both of them at once and `display: none` one of them (and `display: none` the normal one when we're saving it as an image). Only doing it now as there's also a chance that I don't do it this way